Riot/Modules/Common/Avatar/AvatarService.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import MatrixSDK
+import Combine
+import DesignKit
+
+/**
+ Provides a simple api to retrieve and cache avatar images
+ */
+protocol AvatarServiceType {
+    @available(iOS 14.0, *)
+    func avatarImage(inputData: AvatarInputType) -> AnyPublisher<UIImage?, Never>
+}
+
+enum AvatarServiceError: Error {
+    case pathNotfound
+    case loadingImageFailed(Error?)
+}
+
+class AvatarService: AvatarServiceType {
+    
+    private enum Constants {
+        static let mimeType = "image/jpeg"
+        static let thumbnailMethod = MXThumbnailingMethodCrop
+        static let avatarDownloadSize = AvatarSize.xxLarge.size
+    }
+    
+    let avatarGenerator: AvatarGenerator
+    let mediaManager: MXMediaManager
+    
+    init(avatarGenerator: AvatarGenerator, mediaManager: MXMediaManager) {
+        self.avatarGenerator = avatarGenerator
+        self.mediaManager = mediaManager
+    }
+    
+    @available(iOS 14.0, *)
+    func avatarImage(inputData: AvatarInputType) -> AnyPublisher<UIImage?, Never> {
```

Comment:
This could really use a doc comment to mention the generated -> downloaded behaviour.